### PR TITLE
QualifyLeadRequest should update statuscode to Qualified, instead of Open

### LINKED ISSUE #352: QualifyLeadRequest should update statuscode to Qualified, instead of Open
var qualifyReq = new QualifyLeadRequest { 
       CreateAccount = true, CreateContact = true, 
       CreateOpportunity = true, 
       LeadId = lead.ToEntityReference(), 
       Status = new OptionSetValue((int)LeadStatusCode.Qualified), 
    }; 

    var result = service.Execute(qualifyReq); 

result has response and "created entities " but statecode in lead doesn't changed and is still Open. Normally, Statecode is Qualified

Looks like it might be related to [this](https://github.com/jordimontana82/fake-xrm-easy/blob/master/FakeXrmEasy.Shared/FakeMessageExecutors/QualifyLeadRequestExecutor.cs#L101) (guessing it should assign the option set value, not the integer value directly).

@SejoslawMobile  I think you introduced this message... did you have any similar issues?

### PR DESCRIPTION
**Pull request by BetimBeja**
Implemented the solution suggested by @jordimontana82 
Closes #352

### CODE REVIEW

File: FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs [hunk 1]
```diff
@@ -139,5 +139,36 @@ public void Check_if_Account_was_associated_with_Opportunity()
 
             Assert.NotNull(opportunity.CustomerId);
         }
+
+        [Fact]
+        public void Status_of_qualified_Lead_should_be_qualified()
+        {
+            var context = new XrmFakedContext();
+            context.ProxyTypesAssembly = Assembly.GetExecutingAssembly();
+            var service = context.GetFakedOrganizationService();
```
Comment (jordimontana82) [inline FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs]: Hi @BetimBeja , this is amazing. Just a few minor change, GetFakedOrganisationService is deprecated, we should use only GetOrganisationService from now on.

```suggestion
            var service = context.GetOrganizationService();
```

File: FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs [hunk 2]
```diff
@@ -139,5 +139,36 @@ public void Check_if_Account_was_associated_with_Opportunity()
 
             Assert.NotNull(opportunity.CustomerId);
         }
+
+        [Fact]
+        public void Status_of_qualified_Lead_should_be_qualified()
+        {
+            var context = new XrmFakedContext();
+            context.ProxyTypesAssembly = Assembly.GetExecutingAssembly();
```
Comment (jordimontana82) [inline FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs]: There's also no need for this line if early bound entities are passed (Lead) to the Initialize method, it'll discover the assembly for us via reflection :)

```suggestion

```

File: FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs (hunk 1 above)
Comment (BetimBeja) [reply FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs]: I copied another test case and modified it for my specific test and didn't think about this. I guess i should have not been lazy...
Also we should refactor existing tests to not use GetFakedOrganizationService.

Comment (BetimBeja) [reply FakeXrmEasy.Tests.Shared/FakeContextTests/QualifyLeadTests/QualifyLeadTests.cs]: @jordimontana82 , i think you can merge the pull request now 👍

**Comment by jordimontana82** [comment]
Apologies for the delay... was on paternity leave so really didn't have a chance to look into this yet! :)

**Pull request merged**